### PR TITLE
Remove Teradata CREATE args from Snowflake create_sql

### PR DESCRIPTION
**Pull request by treysp**
#1205

### CODE REVIEW

**Comment by treysp** [comment]
I've been meaning to pull most of those args into a new "properties" location so we can use the properties machinery and dialect-specific unsupported approach

File: sqlglot/dialects/snowflake.py [hunk 1]
```diff
@@ -327,3 +327,10 @@ def generatedasidentitycolumnconstraint_sql(
             increment = expression.args.get("increment")
             increment = f" INCREMENT {increment}" if increment else ""
             return f"AUTOINCREMENT{start}{increment}"
+
+        def create_sql(self, expression: exp.Create) -> str:
```
Comment (tobymao) [inline sqlglot/dialects/snowflake.py:331]: delete this and make a new propety location unsupported, then snowflake just needs to override the properties_location

File: sqlglot/dialects/snowflake.py [hunk 2]
```diff
@@ -327,3 +327,10 @@ def generatedasidentitycolumnconstraint_sql(
...
+            unsupported = ["set", "multiset", "global_temporary", "volatile"]
+            for prop in unsupported:
+                expression.set(prop, False)
```
Comment (GeorgeSittas) [inline sqlglot/dialects/snowflake.py:334]: We shouldn't mutate any expression in the generator. Whenever something like this is needed, you first need to copy it.

File: sqlglot/dialects/snowflake.py [hunk 3]
```diff
@@ -327,3 +327,10 @@ def generatedasidentitycolumnconstraint_sql(
...
+            create_sql = super().create_sql(expression)
+            return create_sql
```
Comment (GeorgeSittas) [inline sqlglot/dialects/snowflake.py:336]:
```suggestion
            return super().create_sql(expression)
```

**Pull request closed without merging**